### DISCUSSION TITLE
Change naming convention recommendation?

### DISCUSSION THREAD

**Discussion by jenstornell** [vuejs/pinia#994] [Ideas]
From the docs: `const someStore = useSomeStore()`.
https://pinia.vuejs.org/core-concepts/actions.html#usage-with-the-options-api

 It starts with `use` which is nice but `someStore` ends with  `Store`. I think it would be better to start with it instead, expecially when having multiple store calls.

Compare these two (from my own project).

### `Store` last

```js
const nodeCreateStore     = useNodeCreate()
const shortcutStore       = useShortcuts()
const nodeCollectionStore = useNodeCollection()
const taskCollectionStore = useTaskCollection()
```

### `store` first

```js
const storeNodeCreate     = useNodeCreate()
const storeShortcut       = useShortcuts()
const storeNodeCollection = useNodeCollection()
const storeTaskCollection = useTaskCollection()
```

The last example is much more readable because every `store` is lined up and therefor works like a visual group together.

What do you think?

**Comment by posva** [comment]
I think it makes more sense the other way around, like at https://vuejs.org/v2/style-guide/#Order-of-words-in-component-names-strongly-recommended:

```
Bad

components/
|- ClearSearchButton.vue
|- ExcludeFromSearchInput.vue
|- LaunchOnStartupCheckbox.vue
|- RunSearchButton.vue
|- SearchInput.vue
|- TermsCheckbox.vue


Good
components/
|- SearchButtonClear.vue
|- SearchButtonRun.vue
|- SearchInputQuery.vue
|- SearchInputExcludeGlob.vue
|- SettingsCheckboxTerms.vue
|- SettingsCheckboxLaunchOnStartup.vue
```

**Reply by jenstornell** [reply]
Yeah, well, I actually do a bit of both of these. The first and the last block of code below are my stores. The middle block are my component imports.

```html
<script setup>
// Import stores
import { useDocument }         from 'path'
import { useModal }            from 'path'
import { useNodeCollection }   from 'path'

// Import components
import NodeGroups    from 'path'
import NodeDocuments from 'path'
import NodeCreate    from 'path'

// Const stores
const storeDocument       = useDocument()
const storeModal          = useModal()
const storeNodeCollection = useNodeCollection()
</script>
```

*The code itself may not make sense. It's main goal is to show a name structure convention.*

Do you still find this approach weird?

**Reply by posva** [reply]
You are free to follow which ever conventions works better for you 😄

**Reply by kelvinmatheus** [reply]
@posva, How about compound file names?  Example:

project-statuses.ts
projectStatuses.ts
ProjectStatuses.ts

What is the used standard?

Thanks for your time and attention.
